ibind/base/rest_client.py

```diff
@@ -147,8 +158,18 @@ def request(self, method: str, endpoint: str, attempt: int = 0, log: bool = True
             Exception: For any other errors that occur during the request.
 
         """
-        endpoint = endpoint.lstrip("/")
-        url = f"{self.base_url}{endpoint}"
+        
+
+        # get headers from ibkr_client.py
+        if self._use_oauth:
+            header_oauth=self.get_headers(request_method=method,
+                                          request_url=endpoint,
+                                          request_params=kwargs.get('json'))
+            url=endpoint
+        else:
+            header_oauth=None
+            endpoint = endpoint.lstrip("/")
+            url = f"{self.base_url}{endpoint}"
```

Comment:
No the urls are different.
IBKR OAuth uses ```REQUEST_URL = "https://api.ibkr.com/v1/api/oauth/live_session_token"``` to get the tokens.
If you can see a better method let me know. My aim has been to get the code to work and then improve.
Right now the IBKR code works, I can get the tokens, but the ibind/oauth code isn't working.

Reply:
I see your point, thanks for explaining this.

I can explain the following:

> Right now the IBKR code works, I can get the tokens, but the ibind/oauth code isn't working.

Because you run in `use_oauth=True` mode, you always set `url=endpoint` instead of `url = f"{self.base_url}{endpoint}"`. This works fine for the `live_session_token` endpoint, since you specify the full URL:
```python
def req_live_session_token(self) -> tuple[str, int]:
        ...
        REQUEST_URL = "https://api.ibkr.com/v1/api/oauth/live_session_token"
        ...
        response=self.post(path=REQUEST_URL, params = params, log = True)
```
However if you note the rest of the request methods, they specify only the final few bits of the URL, given that the base remains constant:

```python
    def receive_brokerage_accounts(self: 'IbkrClient') -> Result:  # pragma: no cover
        return self.get('/iserver/accounts')
```

Reply:
There's a way we can get around this. Make `RestClient.request` (and it's wrappers, GET, POST, DELETE) method accept a `full_url` optional argument:

```python
def request(self, method: str, endpoint: str, attempt: int = 0, log: bool = True, full_url: str = None, **kwargs) -> Result:
```

And then we check for it here:

```python
if full_url is None:
    endpoint = endpoint.lstrip("/")
    url = f"{self.base_url}{endpoint}"
else:
    url = full_url
```

This way you can call:

```python
def req_live_session_token(self) -> tuple[str, int]:
        ...
        REQUEST_URL = "https://api.ibkr.com/v1/api/oauth/live_session_token"
        ...
        response=self.post(path=None, full_url=REQUEST_URL, params = params, log = True)
```

Reply:
Thanks for the feedback, the code currently can't successfully establish a session or get market data.
Once the code can successfully get the live session and access tokens (still wip), I'll move on to getting market data.